File: tests/journal.lib/ansi_ansi.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2023 all rights reserved


// get the journal
#include <pyre/journal.h>
// support
#include <cassert>


// type aliases
using csi_t = pyre::journal::csi_t;
using ansi_t = pyre::journal::ansi_t;


// verify that the colors in the ANSI palette are the correct escape sequences
int main() {

    // verify the contents of the {ansi} color table
    // the reset sequence
    assert ((ansi_t::ansi("normal") == csi_t::reset()));
    // regular colors
    assert ((ansi_t::ansi("black") == csi_t::csi3(30)));
    assert ((ansi_t::ansi("red") == csi_t::csi3(31)));
    assert ((ansi_t::ansi("green") == csi_t::csi3(32)));
    assert ((ansi_t::ansi("brown") == csi_t::csi3(33)));
    assert ((ansi_t::ansi("blue") == csi_t::csi3(34)));
    assert ((ansi_t::ansi("purple") == csi_t::csi3(35)));
    assert ((ansi_t::ansi("cyan") == csi_t::csi3(36)));
    assert ((ansi_t::ansi("light-gray") == csi_t::csi3(37)));
    // bright colors
    assert ((ansi_t::ansi("dark-gray") == csi_t::csi3(30, true)));
    assert ((ansi_t::ansi("light-red") == csi_t::csi3(31, true)));
    assert ((ansi_t::ansi("light-green") == csi_t::csi3(32, true)));
    assert ((ansi_t::ansi("yellow") == csi_t::csi3(33, true)));
    assert ((ansi_t::ansi("light-blue") == csi_t::csi3(34, true)));
    assert ((ansi_t::ansi("light-purple") == csi_t::csi3(35, true)));
    assert ((ansi_t::ansi("light-cyan") == csi_t::csi3(36, true)));
    assert ((ansi_t::ansi("white") == csi_t::csi3(37, true)));

    // all done
    return 0;
}


// end of file

File: tests/journal.pkg/ansi_ansi.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2023 all rights reserved


def test():
    """
    Check a few of the canonical ANSI color names
    """
    # access the color map
    from journal.ANSI import ANSI
    # and the control sequence generator
    from journal.CSI import CSI

    # verify the contents of the {ansi} color table
    # the reset sequence
    assert ANSI.ansi("normal") == CSI.reset()

    # regular colors
    assert ANSI.ansi("black") == CSI.csi3(code=30)
    assert ANSI.ansi("red") == CSI.csi3(code=31)
    assert ANSI.ansi("green") == CSI.csi3(code=32)
    assert ANSI.ansi("brown") == CSI.csi3(code=33)
    assert ANSI.ansi("blue") == CSI.csi3(code=34)
    assert ANSI.ansi("purple") == CSI.csi3(code=35)
    assert ANSI.ansi("cyan") == CSI.csi3(code=36)
    assert ANSI.ansi("light-gray") == CSI.csi3(code=37)

    # bright colors
    assert ANSI.ansi("dark-gray") == CSI.csi3(code=30, bright=True)
    assert ANSI.ansi("light-red") == CSI.csi3(code=31, bright=True)
    assert ANSI.ansi("light-green") == CSI.csi3(code=32, bright=True)
    assert ANSI.ansi("yellow") == CSI.csi3(code=33, bright=True)
    assert ANSI.ansi("light-blue") == CSI.csi3(code=34, bright=True)
    assert ANSI.ansi("light-purple") == CSI.csi3(code=35, bright=True)
    assert ANSI.ansi("light-cyan") == CSI.csi3(code=36, bright=True)
    assert ANSI.ansi("white") == CSI.csi3(code=37, bright=True)

    # all done
    return


# main
if __name__ == "__main__":
    # prohibit the journal bindings
    journal_no_libjournal = True
    # run the test
    test()


# end of file
